src/test/java/igrad/logic/commands/AddCommandTest.java

```diff
@@ -193,6 +197,16 @@ public boolean hasRequirement(Requirement requirement) {
             throw new AssertionError("This method should not be called.");
         }
 
+        @Override
+        public Optional<Requirement> getRequirementByName(Name requirementName) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public List<Module> getModulesByModuleCode(List<ModuleCode> moduleCodes) {
+            throw new AssertionError("This method should not be called.");
+        }
+
         @Override
         public void addRequirement(Requirement requirement) {
             throw new AssertionError("This method should not be called.");
```

Comment:
i dont understand the purpose of this class. why do we still have a addcommand class and why are you adding stuff to it? o.O

Reply:
Oh, cos when i try to run the `build` (which compiles all parts of the project it fails, as I've added additional methods to the `Model` Interface)

Yeah, its probably time to clean the tests up (but maybe don't remove for now, I might need  to reference in  future, how these tests are done).

Reply:
oh i see okay!

Reply:
i think  last time, you asked me about what's that build thingy and why do i just need to build it when i could just run  the  app

Reply:
i think build just compiles the whole app